cockpit_api/src/test/java/com/cockpit/api/service/jiragateway/JiraGatewayServiceTest.java

```diff
@@ -69,4 +103,87 @@ public void whenUpdateTotalNbUserStoriesTaskRunsThenUpdateTotalNbUserStoriesInSp
+
+        mockUserStory.setJira(mockJira);
+        mockUserStory.setId(99999L);
+        mockUserStory.setSprint(mockSprint);
+        mockUserStory.setIssueKey("TC");
+        notFoundUserStory.setJira(mockJira);
+        notFoundUserStory.setId(99998L);
+        notFoundUserStory.setSprint(mockSprint);
+        notFoundUserStory.setIssueKey("NU");
+
+
+        List<UserStory> mockUserStoryList = new ArrayList<>();
+        mockUserStoryList.add(mockUserStory);
+        List<UserStory> notFoundUserStoryList = new ArrayList<>();
+        notFoundUserStoryList.add(notFoundUserStory);
+
+        List<Sprint> mockSprintList = new ArrayList<>();
+        mockSprintList.add(mockSprint);
+        List<Sprint> notFoundSprintList = new ArrayList<>();
+        notFoundSprintList.add(notFoundSprint);
+
+        List<Jira> mockJiraList = new ArrayList<>();
+        mockJiraList.add(mockJira);
+        List<Jira> notFoundJiraList = new ArrayList<>();
+        notFoundJiraList.add(notFoundJira);
+
+        // TEST JIRA DELETE SCHEDULED TASK
+        JSONArray notFoundJiraJsonArray = new JSONArray();
+        for (Jira jira : notFoundJiraList) {
+            notFoundJiraJsonArray.put(jira);
+        }
+        JiraGatewayService mockJiraGatewayService = mock(JiraGatewayService.class);
+        ReflectionTestUtils.setField(jiraGatewayService, "username", username);
+        ReflectionTestUtils.setField(jiraGatewayService, "token", token);
+        ReflectionTestUtils.setField(jiraGatewayService, "jiraUrl", jiraUrl);
+        Mockito.when(jiraRepository.findAllByOrderById()).thenReturn(mockJiraList);
+        Mockito.when(mockJiraGatewayService.getJiraProjects()).thenReturn(notFoundJiraJsonArray);
+        jiraGatewayService.deleteJiraProjects();
+        assertThat(jiraRepository.findByJiraProjectKey(mockJira.getJiraProjectKey())).isNull();
+
+        // TEST SPRINT DELETE SCHEDULED TASK
+        Mockito.when(sprintRepository.findAll()).thenReturn(mockSprintList);
+        JSONArray notFoundSprintJsonArray = new JSONArray();
+        for (Sprint sprint : notFoundSprintList) {
+            notFoundSprintJsonArray.put(sprint);
+        }
+        Mockito.when(mockJiraGatewayService.getJiraSprints(anyLong())).thenReturn(notFoundSprintJsonArray);
+        jiraGatewayService.deleteJiraSprint();
+        assertThat(sprintRepository.findByJiraSprintId(notFoundSprint.getJiraSprintId())).isNull();
+
+        // TEST USER STORY DELETE SCHEDULED TASK
+        Mockito.when(userStoryRepository.findAll()).thenReturn(mockUserStoryList);
+        JSONArray notFoundUserStoryJsonArray = new JSONArray();
+        for (UserStory userStory : notFoundUserStoryList) {
+            notFoundUserStoryJsonArray.put(userStory);
+        }
+        Mockito.when(mockJiraGatewayService.getJiraIssueList()).thenReturn(notFoundUserStoryJsonArray);
+        jiraGatewayService.deleteJiraIssues();
+        assertThat(userStoryRepository.findByIssueKey(notFoundUserStory.getIssueKey())).isNull();
```

Comment:
same as before, when i put         assertThat(userStoryRepository.findByIssueKey(notFoundUserStory.getIssueKey())).isNull();
before jiraGatewayService.deleteJiraIssues();
it passes, too, is it normal ?